esp-hal-common/src/embassy/mod.rs

```diff
@@ -96,7 +98,70 @@ use time_driver::EmbassyTimer;
 
 use crate::clock::Clocks;
 
+/// Initialise embassy, including setting up interrupts for the DMA and async
+/// enabled peripherals.
 pub fn init(clocks: &Clocks, td: time_driver::TimerType) {
+    #[cfg(any(esp32s3, esp32c6, esp32h2))]
+    crate::interrupt::enable(Interrupt::DMA_IN_CH0, Priority::max()).unwrap();
+    #[cfg(any(esp32s3, esp32c6, esp32h2))]
+    crate::interrupt::enable(Interrupt::DMA_OUT_CH0, Priority::max()).unwrap();
+    #[cfg(any(esp32s3, esp32c6, esp32h2))]
+    crate::interrupt::enable(Interrupt::DMA_IN_CH1, Priority::max()).unwrap();
+    #[cfg(any(esp32s3, esp32c6, esp32h2))]
+    crate::interrupt::enable(Interrupt::DMA_OUT_CH1, Priority::max()).unwrap();
+    #[cfg(any(esp32s3, esp32c6, esp32h2))]
+    crate::interrupt::enable(Interrupt::DMA_IN_CH2, Priority::max()).unwrap();
+    #[cfg(any(esp32s3, esp32c6, esp32h2))]
+    crate::interrupt::enable(Interrupt::DMA_OUT_CH2, Priority::max()).unwrap();
+
+    #[cfg(esp32s3)]
+    crate::interrupt::enable(Interrupt::DMA_IN_CH3, Priority::max()).unwrap();
+    #[cfg(esp32s3)]
+    crate::interrupt::enable(Interrupt::DMA_OUT_CH3, Priority::max()).unwrap();
+
+    #[cfg(any(esp32c3, esp32c2))]
+    crate::interrupt::enable(Interrupt::DMA_CH0, Priority::max()).unwrap();
+    #[cfg(esp32c3)]
+    crate::interrupt::enable(Interrupt::DMA_CH1, Priority::max()).unwrap();
+    #[cfg(esp32c3)]
+    crate::interrupt::enable(Interrupt::DMA_CH2, Priority::max()).unwrap();
+
+    #[cfg(any(esp32))]
+    crate::interrupt::enable(Interrupt::SPI1_DMA, Priority::max()).unwrap();
+    #[cfg(any(esp32, esp32s2))]
+    crate::interrupt::enable(Interrupt::SPI2_DMA, Priority::max()).unwrap();
+    #[cfg(any(esp32, esp32s2))]
+    crate::interrupt::enable(Interrupt::SPI3_DMA, Priority::max()).unwrap();
+    #[cfg(esp32s2)]
+    crate::interrupt::enable(Interrupt::SPI4_DMA, Priority::max()).unwrap();
+
+    // see https://github.com/esp-rs/esp-pacs/issues/175
```

Comment:
I've published new PACs for the ESP32-C6 and ESP32-H2, so this interrupt has now been renamed.

Reply:
Thanks, this should just need a rebase once the pacs are updated in esp-hal.